src/Workspaces/Core/Portable/Remote/RemoteHostSessionHelpers.cs

```diff
@@ -113,19 +113,20 @@ public void Shutdown()
                     _client.StatusChanged -= OnStatusChanged;
                 }
 
-                connection = _lazyConnection;
+                connection = _lazyConnection.Move();
 
                 _client = null;
-                _lazyConnection = null;
+                _lazyConnection = default;
             }
 
-            connection?.Dispose();
+            if (!connection.IsDefault)
+                connection.Dispose();
```

Comment:
My proposal for non-defaultable value types does not add support for `?.`, so this is expanded to the full form. It may be possible to improve the experience from the API side by (somehow) declaring the `Dispose()` method to allow default instance.